ui/src/components/crop_image_tool/rectangle_format_tool/mod.rs

```diff
@@ -0,0 +1,212 @@
+                            appearance: Appearance::DangerAlternative,
+                            icon: Shape::XMark,
+                            onpress: move |_| {
+                                cx.props.on_cancel.call(());
+                                crop_image.set(false);
+                            }
+                        },
+                        div {
+                            margin_right: "16px",
+                        }
+                        Button {
+                            appearance: Appearance::Success,
+                            icon: Shape::Check,
+                            onpress: move |_| {
+                                cx.spawn({
+                                    to_owned![eval, image_scale, cropped_image_pathbuf, clicked_button_to_crop];
+                                    async move {
+                                        let save_image_cropped_js = SAVE_CROPPED_IMAGE_SCRIPT
+                                        .replace("$IMAGE_SCALE", (1.0 / *image_scale.read()).to_string().as_str());
+                                        if let Ok(r) = eval(&save_image_cropped_js) {
+                                            if let Ok(val) = r.join().await {
+                                                let thumbnail = val.as_str().unwrap_or_default();
+                                                let base64_string = thumbnail.trim_matches('\"');
+                                                let decoded_bytes = match base64::decode(base64_string) {
+                                                    Ok(bytes) => bytes,
+                                                    Err(e) => {
+                                                        log::error!("Error decoding base64 string for cropped image: {}", e);
+                                                        return;
+                                                    },
+                                                };
+                                                let cropped_image_path = STATIC_ARGS.uplink_path.join("cropped_image_for_banner.png");
+                                                let mut file = match tokio::fs::File::create(cropped_image_path.clone()).await {
+                                                    Ok(file) => file,
+                                                    Err(e) => {
+                                                        log::error!("Error creating cropped image file: {}", e);
+                                                        return;
+                                                    },
+                                                };
+
+                                                if let Err(e) = file.write_all(&decoded_bytes).await {
+                                                    log::error!("Error writing cropped image file. {}", e);
+                                                    return;
+                                                }
```

Comment:
If you are to write to a file instead of passing the bytes of data along, you should make sure to use `file.sync_all().await` to make sure the data has written to disk.